tests/lib/rules/no-implicit-globals.js

```diff
@@ -0,0 +1,233 @@
+        },
+        {
+            code: "window.foo = function() {};"
+        },
+        {
+            code: "window.foo = function foo() {};"
+        },
+        {
+            code: "self.foo = 1;"
+        },
+        {
+            code: "self.foo = function() {};"
+        },
+        {
+            code: "this.foo = 1;"
+        },
+        {
+            code: "this.foo = function() {};"
+        },
+        {
+            code: "Utils.foo = 1;"
+        },
+        {
+            code: "Utils.foo = function() {};"
+        },
+        {
+            code: "(function() { var foo = 1; })();"
+        },
+        {
+            code: "(function() { let foo = 1; })();",
+            ecmaFeatures: { blockBindings: true }
+        },
+        {
+            code: "(function() { const foo = 1; })();",
+            ecmaFeatures: { blockBindings: true }
+        },
+        {
+            code: "(function() { function foo() {} })();"
+        },
+        {
+            code: "foo = 1;",
+            ecmaFeatures: { modules: true }
+        },
+        {
+            code: "var foo = 1;",
+            ecmaFeatures: { modules: true }
+        },
+        {
+            code: "let foo = 1;",
+            ecmaFeatures: { modules: true, blockBindings: true }
+        },
+        {
+            code: "const foo = 1;",
+            ecmaFeatures: { modules: true, blockBindings: true }
+        },
+        {
+            code: "function foo() {}",
+            ecmaFeatures: { modules: true }
+        },
+        {
+            code: "var foo = 1;",
+            ecmaFeatures: { globalReturn: true }
+        },
+        {
+            code: "let foo = 1;",
+            ecmaFeatures: { globalReturn: true, blockBindings: true }
+        },
+        {
+            code: "const foo = 1;",
+            ecmaFeatures: { globalReturn: true, blockBindings: true }
+        },
+        {
+            code: "function foo() {}",
+            ecmaFeatures: { globalReturn: true }
+        }
+    ],
+
+    invalid: [
+        {
+            code: "foo = 1;",
+            errors: [
+                {
+                    message: "Implicit global variable, assign as window property instead.",
+                    type: "AssignmentExpression"
+                }
+            ]
+        },
+        {
+            code: "foo = 1, bar = 2;",
+            errors: [
+                {
+                    message: "Implicit global variable, assign as window property instead.",
+                    type: "AssignmentExpression"
+                },
+                {
+                    message: "Implicit global variable, assign as window property instead.",
+                    type: "AssignmentExpression"
+                }
+            ]
+        },
+        {
+            code: "var foo = 1;",
+            errors: [
+                {
+                    message: "Implicit global variable, assign as window property instead.",
+                    type: "VariableDeclarator"
```

Comment:
I wasn't sure what nodes would be conventional attaching the errors to. This seem okay?